### ISSUE TITLE
Suggestions for improving the product compare feature in jet cw

### ISSUE THREAD

**Issue by 79ho3ein** [CrocoBlock/suggestions#7487]
Hi crocoblock team,

1- border-radius for <th> and <td> : https://prnt.sc/6gQoctaYvOxs
2- Custom icon for add to cart button and more info button in variable products : https://prnt.sc/mIKprwMbVkhB
3- Custom message "Could not add more numbers to comparison list". Currently, **if the user adds more than 4 products to the comparison list, this message is displayed as a popup and there is no option to customize.**
4- If empty, display custom text. for example - : https://prnt.sc/pVUKjrMCUWV6
5- Title length option for the product title in the comparison list (such as the product title in the product grid widget)
6- Option to control product image size as percentage, px or...
7- "Inline" or "block" mode for product prices (such as product prices in the product grid widget)
8- Currently, if there are no products in the comparison list, a plain text is displayed. It would be great if you could add this option so that we can use the shortcode instead of its text so that we can display a custom template for it. Just like the product grid widget.

cheers

**Comment by UraraReika** [comment]
1 - We will not add the border radius controls for table cells will have some style conflict with borders because compare table use collapse.
2 - This one will not be added as well. If there will be enough requests this can be added to https://github.com/UraraReika/jet-woo-builder-custom-add-to-cart-icon.
3 - You can change this message in plugin settings - https://tppr.me/TKBVho. If you want to change styles then use CSS, controls will not be added.
4 - Each elements has hooks that can help you to implement unique text for empty item. As for the attributes row in the next release you can use `jet-cw/widgets/compare/empty-item-fallback` hook to customize it as well. Like this:
```php
add_filter('jet-cw/widgets/compare/empty-item-fallback', function () {
	return __( 'Empty item fallback', 'jet-cw' );
});
```
5 - Since the title is also part of the comparison and may contain unique information, adding this functionality is not appropriate, so it will not be done.
6 - Width and Max Width controls will be added in the next release for Compare Table widget thumbnails.
7 - This one although will not be added.
8 - You already can do it the same way as in the product grid/list widgets. With the help of this snippet - https://gist.github.com/UraraReika/e9f09e16c678e78a6f9bfc85fc0d765a. Correct hook in the comment - `jet-cw/compare/empty_text`.

**Comment by Crocoblock** [comment]
was fixed and released in 1.5.7
